controllers/spiaccesstoken_controller.go

```diff
@@ -165,6 +166,25 @@ func (r *SPIAccessTokenReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return ctrl.Result{}, nil
 	}
 
+	// cleanup tokens which are in awaiting state and have their parent binging removed.
+	// grace period taken in account to avoid race condition during the object creation process (i.e. when token is created but not yet linked)
+	if time.Since(at.CreationTimestamp.Time).Seconds() > NoLinkingBindingGracePeriodSeconds && at.Status.Phase == api.SPIAccessTokenPhaseAwaitingTokenData {
```

Comment:
! nitpick alert ! `time.Since(at.CreationTimestamp.Time).Seconds() > NoLinkingBindingGracePeriodSeconds` is much more expensive than `at.Status.Phase == api.SPIAccessTokenPhaseAwaitingTokenData`. I would swap them, so more expensive condition has chance to not be evaluated.

Reply:
ok